### PR TITLE
fixes #5795

### LINKED ISSUE #5795: Linenoise returned nil [IOError]
```nim
import terminal,rdstdin

proc exitHandler() {.noconv.} =
    echo("Have a nice day")
    addQuitProc(resetAttributes)
    quit(0)
   
setControlCHook(exitHandler)  

var z = readLineFromStdin("Write something or press Ctrl-C to exit : ")
echo z
 
```

pressing Ctrl-C throws error rather than expected output of exitHandler

```nim
 rdstdin.nim(109)         readLineFromStdin
 Error: unhandled exception: Linenoise returned nil [IOError]
```

### PR DESCRIPTION
**Pull request by krux02**
fixes https://github.com/nim-lang/Nim/issues/5795

### CODE REVIEW

File: lib/impure/rdstdin.nim [hunk 1]
```diff
@@ -100,12 +100,17 @@ when defined(Windows):
...
 else:
-  import linenoise, termios
+  import linenoise, termios, posix
 
   proc readLineFromStdin*(prompt: string): TaintedString {.
                           tags: [ReadIOEffect, WriteIOEffect].} =
+    errno = 0
     var buffer = linenoise.readLine(prompt)
     if isNil(buffer):
+      # linenoise eats SIGINT and sets errno to EAGAIN, therefore reraise SIGINT.
```
Comment (Araq) [inline lib/impure/rdstdin.nim]: So patch linenoise. No more ``errno`` nonsense please.

File: lib/impure/rdstdin.nim [hunk 2]
```diff
@@ -100,12 +100,17 @@ when defined(Windows):
     stdout.write "\n"
 
 else:
-  import linenoise, termios
+  import linenoise, termios, posix
 
   proc readLineFromStdin*(prompt: string): TaintedString {.
                           tags: [ReadIOEffect, WriteIOEffect].} =
+    errno = 0
```
Comment (Araq) [inline lib/impure/rdstdin.nim]: Errno can be resolved to a function call on some targets, it should not be set directly. Also 'errno' is for posix, not for user level code.

Comment (krux02) [inline lib/impure/rdstdin.nim]: https://linux.die.net/man/3/errno
> _errno_ is defined by the ISO C standard to be a modifiable lvalue of type _int_ [...]; _errno_ may be a macro
Therefore writing to it is just fine, even necessary to get correct error codes.

File: lib/impure/rdstdin.nim (hunk 1 above)
Comment (krux02) [inline lib/impure/rdstdin.nim]: I did not come up with ``errno`` not did I come up with this API. I also did not come up with the idea to use linenoise. All I did was to make this API work with what linenoise provides. Don't blame me for the bad design of the linenoise library.

**Pull request closed without merging**